core/org.eclipse.cdt.core.native/src/org/eclipse/cdt/utils/spawner/ProcessFactory.java

```diff
@@ -81,8 +176,74 @@ public Builder gracefulExitTimeMs(int gracefulExitTimeMs) {
 			return this;
 		}
 
+		private StringBuilder debug() {
+			/*
+			 * for debug purpose
+			 */
+			StringBuilder sb = new StringBuilder();
+
+			sb.append("command :\n"); //$NON-NLS-1$
+			for (int i = 0; i < cmdarray.length; i++) {
+				sb.append(i);
+				sb.append(" : \""); //$NON-NLS-1$
+				sb.append(cmdarray[i]);
+				sb.append("\"\n"); //$NON-NLS-1$
+			}
+			sb.append("\n\n"); //$NON-NLS-1$
+
+			sb.append("directory :\n"); //$NON-NLS-1$
+			if (dir != null) {
+				String pathString = dir.toString();
+				sb.append(pathString);
+				sb.append('\n');
+				Path path = new Path(pathString);
+				String pathDevice = path.getDevice();
+				String[] pathSegments = path.segments();
+				if (pathDevice != null) {
+					sb.append("device : "); //$NON-NLS-1$
+					sb.append(pathDevice);
+					sb.append('\n');
+				}
+				sb.append("segments : \n"); //$NON-NLS-1$
+				for (int i = 0; i < pathSegments.length; i++) {
+					sb.append(i);
+					sb.append(" : "); //$NON-NLS-1$
+					sb.append(pathSegments[i]);
+					sb.append('\n');
+				}
+			} else {
+				sb.append("not specified\n"); //$NON-NLS-1$
+			}
+			sb.append("\n\n"); //$NON-NLS-1$
+
+			{
+				TreeMap<String, String> environmentA = getDefaultEnvironment();
+				TreeMap<String, String> environmentB = envpToEnvMap(envp);
+
+				sb.append("environment :\n"); //$NON-NLS-1$
+				appendEnvMapComparison(sb, environmentA, environmentB);
+				sb.append("\n\n"); //$NON-NLS-1$
+			}
+			if (use_pty) {
+				sb.append("use pty : "); //$NON-NLS-1$
+				sb.append(pty.toString());
+				sb.append("\n\n"); //$NON-NLS-1$
+			}
+			if (has_gracefulExitTimeMs) {
+				sb.append("has gracefulExitTimeMs : "); //$NON-NLS-1$
+				sb.append(gracefulExitTimeMs);
+				sb.append("\n\n"); //$NON-NLS-1$
+			}
+			/*
+			 * set breakpoint on next line to inspect sb when debugging, to see the
+			 * ultimate parameters of ProcessBuilder
+			 */
+			return sb;
+		}
+
 		public Process start() throws IOException {
 			cmdarray = modifyCmdArrayIfFlatpak(cmdarray);
+			debug();
```

Comment:
I am not keen on having this run all the time. Can you consider putting this behind a trace flag or simply commenting out this line?

```suggestion
			// Uncomment the next line, step over debug() to get detailed info on what is being launched.
			// var debug = debug();
```